fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java

```diff
@@ -630,12 +626,11 @@ private VehicleSensorData rearDistance(double rearDistance) {
     /**
      * Calculates the sensor data from rear.
      *
-     * @param bridge Connection to Traci.
      * @param subscriptions   Subscription data.
      * @return Calculated sensor data.
      * @throws InternalFederateException if leading vehicle for a vehicle couldn't be read
      */
-    private Map<String, VehicleSensorData> calculateBackSensorData(Bridge bridge, List<AbstractSubscriptionResult> subscriptions) throws InternalFederateException {
+    private Map<String, VehicleSensorData> calculateBackSensorData(List<AbstractSubscriptionResult> subscriptions) {
```

Comment:
It is not clear why the method is called `calculateBackSensorData()` when also `vehiclesWithFrontSensor` are taken into account. 

It seems that this method "calculates" (or better determines) leading vehicles globally and the later called `calculateSensorData()` prepares the global data for each individual vehicle??
Actually, the whole `SimulationFacade` is already quite long and hard to understand for non-insiders. Maybe further partitions would be helpful.

Reply:
SUMO can only return the leader of a vehicle (i.e. front sensor). this method uses the front sensor data of all vehicles to calculate the back sensor data (a vehicle is the follower of its leader). I guess a better comment would be sufficient.

Reply:
But yes, apparently this also calculates the front distance sensor based on leader information, I'll have a look how this method can be renamed.

Reply:
I refactored the code quite a bit. Decreased complexity and increased readability 🥳